### ISSUE TITLE
import/export: false positives

### ISSUE THREAD

**Issue by JamesHenry** [typescript-eslint/typescript-eslint#10]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/444

---

**What version of TypeScript are you using?**
2.6.1

**What version of `typescript-eslint-parser` are you using?**
12.0.0

**What code were you trying to parse?**

```ts
// unexected error: Multiple exports of name 'Foo'. (import/export)
export const Foo = 1;
// unexected error: Multiple exports of name 'Foo'. (import/export)
export type Foo = 1;
```

The import/export rule is defined here: https://github.com/benmosher/eslint-plugin-import/blob/219a8d2e5af230f73b4754f725916843645889ff/docs/rules/export.md

Is this an issue to log here or with the eslint-plugin-import project?

We've decided to disable this ESLint rule as TypeScript also provides similar functionality anyway, but logging this just for others/reference.

For context, this pattern is common when using [unionize](https://github.com/pelotom/unionize):

``` ts
import { unionize, ofType } from 'unionize'

export const Action = unionize({
  ADD_TODO:                ofType<{ id: string; text: string }>(),
  SET_VISIBILITY_FILTER:   ofType<'SHOW_ALL' | 'SHOW_ACTIVE' | 'SHOW_COMPLETED'>(),
  TOGGLE_TODO:             ofType<{ id: string }>(),
});

export type Action = typeof Action._Union;
```

**Comment by j-f1** [comment]
I think this was incorrectly closed.

**Comment by bradzacher** [comment]
@armano2 this will be fixed by your scope analysis changes wouldn't it?

**Comment by RIP21** [comment]
Same with d.ts with many exports inside of the modules.

![image](https://user-images.githubusercontent.com/3940079/54394860-638dc500-46ae-11e9-9ef2-82eb1d66031a.png)

**Comment by bradzacher** [comment]
is an issue for eslint-plugin-import

Reported at https://github.com/benmosher/eslint-plugin-import/issues/1318

----

The namespace bug is separately (already) reported at https://github.com/benmosher/eslint-plugin-import/issues/1300

**Comment by bradzacher** [comment]
FYI - both issues benmosher/eslint-plugin-import#1318 and benmosher/eslint-plugin-import#1300 are now both fixed pending release.
Keep an eye out for the next version of eslint-plugin-import
